src/pytest_bdd/gherkin_terminal_reporter.py

```diff
@@ -66,16 +67,27 @@ def pytest_runtest_logreport(self, report: TestReport) -> Any:
             word_markup = {"yellow": True}
         feature_markup = {"blue": True}
         scenario_markup = word_markup
+        rule_markup = {"purple": True}
 
         if self.verbosity <= 0 or not hasattr(report, "scenario"):
             return super().pytest_runtest_logreport(rep)
 
+        rule = report.scenario.get("rule")
+        indent = "    " if rule else ""
+
         if self.verbosity == 1:
             self.ensure_newline()
             self._tw.write(f"{report.scenario['feature']['keyword']}: ", **feature_markup)
             self._tw.write(report.scenario["feature"]["name"], **feature_markup)
             self._tw.write("\n")
-            self._tw.write(f"    {report.scenario['keyword']}: ", **scenario_markup)
+
+            if rule and rule["name"] != self.current_rule:
+                self._tw.write(f"  {rule['keyword']}: ", **rule_markup)
+                self._tw.write(rule["name"], **rule_markup)
+                self._tw.write("\n")
+                self.current_rule = rule["name"]
```

Comment:
Can you make sure these are covered by tests?

Reply:
Added different verbosity levels to the existing test